localstack/services/awslambda/invocation/runtime_executor.py

```diff
@@ -0,0 +1,233 @@
+    )
+
+
+def get_code_path_for_function(function_version: FunctionVersion) -> Path:
+    return get_path_for_function(function_version) / "code"
+
+
+def get_image_name_for_function(function_version: FunctionVersion) -> str:
+    return f"localstack/lambda-{function_version.id.qualified_arn().replace(':', '_').replace('$', '_').lower()}"
+
+
+def get_image_for_runtime(runtime: str) -> str:
+    runtime, version = get_runtime_split(runtime)
+    return f"{IMAGE_PREFIX}{runtime}:{version}"
+
+
+def get_runtime_client_path() -> Path:
+    return Path(LAMBDA_RUNTIME_INIT_PATH)
+
+
+def prepare_image(target_path: Path, function_version: FunctionVersion) -> None:
+    if not function_version.config.runtime:
+        LOG.error("Images without runtime are currently not supported")
+        raise Exception("Custom images are currently not supported")
+    src_init = get_runtime_client_path()
+    # copy init file
+    target_init = target_path / "aws-lambda-rie"
+    shutil.copy(src_init, target_init)
+    target_init.chmod(0o755)
+    # copy code
+    # create dockerfile
+    docker_file_path = target_path / "Dockerfile"
+    docker_file = LAMBDA_DOCKERFILE.format(
+        base_img=get_image_for_runtime(function_version.config.runtime),
+        rapid_entrypoint=RAPID_ENTRYPOINT,
+    )
+    with docker_file_path.open(mode="w") as f:
+        f.write(docker_file)
+    try:
+        CONTAINER_CLIENT.build_image(
+            dockerfile_path=str(docker_file_path),
+            image_name=get_image_name_for_function(function_version),
+        )
+    except Exception as e:
+        LOG.error("Exception: %s", e)
+
+
+def prepare_version(function_version: FunctionVersion) -> None:
+    if not function_version.code.zip_file:
+        LOG.error("Images without zip_file are currently not supported")
+        raise Exception("Custom images are currently not supported")
+    time_before = time.perf_counter()
+    target_path = get_path_for_function(function_version)
+    target_path.mkdir(parents=True, exist_ok=True)
+    # write code to disk
+    target_code = get_code_path_for_function(function_version)
+    with NamedTemporaryFile() as file:
+        file.write(function_version.code.zip_file)
+        file.flush()
+        unzip(file.name, str(target_code))
+    if config.LAMBDA_PREBUILD_IMAGES:
+        prepare_image(target_path, function_version)
+    LOG.debug("Version preparation took %0.2fms", (time.perf_counter() - time_before) * 1000)
+
+
+def cleanup_version(function_version: FunctionVersion) -> None:
+    function_path = get_path_for_function(function_version)
+    try:
+        shutil.rmtree(function_path)
+    except OSError as e:
+        LOG.debug(
+            "Could not cleanup function %s due to error %s while deleting file %s",
+            function_version.qualified_arn,
+            e.strerror,
+            e.filename,
+        )
```

Comment:
could simplify this with `localstack.utils.files.rm_rf(function_path)`

Reply:
Not a big fan of this method, especially if I only need to remove a folder, but I will use the convention :)